### PR TITLE
[MM-3274] Changed ctrl+/ detection to event.key before event.keyCode

### PR DESCRIPTION
**Pull request by gnufede**
#### Summary
Changed `ctrl+/` shortcut detection to rely on event.key instead of event.keyCode, as the latter isn't reliable across different keyboard layouts.

Also, note that event.keyCode is deprecated [https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode)

#### Ticket Link
Github issue: [https://github.com/mattermost/mattermost-server/issues/7665](https://github.com/mattermost/mattermost-server/issues/7665)
Jira ticket: [https://mattermost.atlassian.net/browse/MM-3274](https://mattermost.atlassian.net/browse/MM-3274)

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: tests/utils/utils.test.jsx [hunk 1]
```diff
@@ -531,3 +531,34 @@ describe('Utils.isEmail', function() {
...
+            {
+                event: new KeyboardEvent('keydown', {key: '/', keyCode: 55}),
+                key: ['/', 191],
+                valid: true,
+            },
+            {
+                event: new KeyboardEvent('keydown', {key: 'ù', keyCode: 191}),
+                key: ['/', 191],
+                valid: false,
+            },
```
Comment (saturninoabril) [inline tests/utils/utils.test.jsx]: Could you add test case where `event.key` is undefined and Unidentified?

**Comment by gnufede** [comment]
While working on this PR that solves just one shortcut for non-US keyboard layouts, I found out that we're using KeyboardEvent.keyCode, that is [deprecated](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode). If you like the approach of the PR, I could apply it project-wide so we avoid having problems with non-US keyboards and stop relying on deprecated properties

File: components/create_post/create_post.jsx [hunk 2]
```diff
@@ -557,7 +557,8 @@ export default class CreatePost extends React.Component {
     }
 
     showShortcuts(e) {
-        if ((e.ctrlKey || e.metaKey) && e.keyCode === KeyCodes.FORWARD_SLASH) {
+        const FORWARD_SLASH_KEY = ['/', 191];
```
Comment (saturninoabril) [inline components/create_post/create_post.jsx]: (Just a note)
If we'll be expanding support to keys for all equivalent at `Contants.KeyCodes`, then we might move this to Constants in the future.

**Comment by jasonblais** [comment]
@gnufede 

> While working on this PR that solves just one shortcut for non-US keyboard layouts, I found out that we're using KeyboardEvent.keyCode, that is deprecated. If you like the approach of the PR, I could apply it project-wide so we avoid having problems with non-US keyboards and stop relying on deprecated properties

We'd be open for a PR - we'd just have to test it carefully that there aren't any breaking changes from the existing behaviour.

File: tests/utils/utils.test.jsx [hunk 3]
```diff
@@ -554,9 +554,78 @@ describe('Utils.isKeyPressed', function() {
...
+                key: ['not-used-field', 2220],
+                valid: true,
+            },
+            {
+                event: new KeyboardEvent('keydown', {key: 'Dead', keyCode: 2220}),
+                key: [null, 2220],
+                valid: true,
+            },
+            {
+                event: new KeyboardEvent('keydown', {key: 'Dead', keyCode: 2220}),
```
Comment (saturninoabril) [inline tests/utils/utils.test.jsx]: Thanks! Last one, I guess the last 3 events above should be `{key: 'Unidentified'`.

File: tests/utils/utils.test.jsx [hunk 4]
```diff
@@ -554,9 +554,78 @@ describe('Utils.isKeyPressed', function() {
...
+                key: ['not-used-field', 2221],
+                valid: true,
+            },
+            {
+                event: {key: undefined, keyCode: 2221},
+                key: [null, 2221],
+                valid: true,
+            },
+            {
+                event: {key: undefined, keyCode: 2221},
```
Comment (saturninoabril) [inline tests/utils/utils.test.jsx]: Build failed due to the use of `undefined` here. Either disable eslint rule to this particular line or just do like: `event: {keyCode: 2221},`
Please run `make check-style` on your local dev as well.

**Comment by hmhealey** [comment]
Looks like the issues @saturninoabril raised may not be fixed, so I'll let him clear that label

**Comment by jwilander** [comment]
Thanks @gnufede !

**Pull request merged**